app/assets/javascripts/discourse/app/models/user.js

```diff
@@ -130,15 +130,57 @@ export function addSaveableUserOptionField(fieldName) {
   userOptionFields.push(fieldName);
 }
 
+function userOption(userOptionKey) {
+  return computed(`user_option.${userOptionKey}`, {
+    get(key) {
+      deprecated(
+        `Getting ${key} property of user object is deprecated. Use user_option object instead`,
+        { id: "discourse.user.userOptions" }
+      );
+
+      return this.get(`user_option.${key}`);
+    },
+
+    set(key, value) {
+      deprecated(
+        `Setting ${key} property of user object is deprecated. Use user_option object instead`,
+        { id: "discourse.user.userOptions" }
+      );
+
+      if (!this.user_option) {
+        this.set("user_option", {});
+      }
+
+      return this.set(`user_option.${key}`, value);
+    },
+  });
+}
+
 const User = RestModel.extend({
+  mailing_list_mode: userOption("mailing_list_mode"),
```

Comment:
Could we lean on just iterating through the existing `userOptionFields` here for this? Since we also have to take into account `addSaveableUserOptionField` and make sure that still works

Reply:
I do see we modified `filteredUserOptionFields`, we should just double check any plugins relying on this API are still working

Reply:
I wanted to iterate over `userOptionFields`, but I don't think that is a good approach because not all user options are accessed via the user object directly. I added aliases just for the ones that were serialized by `CurrentUserSerializer`.

`filteredUserOptionFields` has not been modified, it holds the same values as before.